device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -705,10 +722,6 @@ private void openConnection() throws TransportException
 
         //Codes_SRS_IOTHUBTRANSPORT_34_039: [This function shall open the iotHubTransportConnection object with the saved list of configs.]
         this.iotHubTransportConnection.open(this.deviceClientConfigs, scheduledExecutorService);
-
-        //Codes_SRS_IOTHUBTRANSPORT_34_040: [This function shall invoke the method updateStatus with status CONNECTED,
-        // reason CONNECTION_OK, and a null throwable.]
-        this.updateStatus(IotHubConnectionStatus.CONNECTED, IotHubConnectionStatusChangeReason.CONNECTION_OK, null);
```

Comment:
Will this never be triggered now? Are we changing the state diagram for application?

Reply:
The status update for CONNECTED will now come only from the onConnectionEstablished callback within the same class. Previously, we had this line here only because the Http layer would never call onConnectionEstablished. However, as a result, there would be two calls to updateStatus with CONNECTED around the same time for MQTT and AMQP users. 

I simplified this by making the http layer just call onConnectionEstablished like MQTT and AMQP do.